src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/SearchParameterFilterAttribute.cs

```diff
@@ -0,0 +1,67 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Hl7.Fhir.Model;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Health.Fhir.Api.Features.Routing;
+using Microsoft.Health.Fhir.Shared.Core.Features.Search.Parameters;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Api.Features.Filters
+{
+    [AttributeUsage(AttributeTargets.Method)]
+    internal class SearchParameterFilterAttribute : ActionFilterAttribute
+    {
+        private ISearchParameterValidator _searchParameterValidator;
+        private ISearchParameterEditor _searchParameterEditor;
+
+        public SearchParameterFilterAttribute(ISearchParameterValidator searchParamValidator, ISearchParameterEditor searchParameterEditor)
+        {
+            EnsureArg.IsNotNull(searchParamValidator, nameof(searchParamValidator));
+            EnsureArg.IsNotNull(searchParameterEditor, nameof(searchParameterEditor));
+
+            _searchParameterValidator = searchParamValidator;
+            _searchParameterEditor = searchParameterEditor;
+        }
+
+        public override async Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+        {
+            EnsureArg.IsNotNull(context, nameof(context));
+
+            if (context.ActionArguments.TryGetValue(KnownActionParameterNames.Resource, out var parsedModel))
+            {
+                if (parsedModel is SearchParameter)
+                {
+                    // wait for the validation checks to pass before allowing the FHIRController action to continue
+                    await _searchParameterValidator.ValidateSearchParamterInput(
+                        parsedModel as SearchParameter,
```

Comment:
>parsedModel as SearchParameter [](start = 24, length = 30)

nit:
you can do instead of 
```
if (parsedModel is SearchParameter)
```
this:
```
if (parsedModel is SearchParameter searchParameter)
```
and use `searchParameter` variable instead of `parsedModel as SearchParameter` here and few lines below.

Reply:
Updated.